drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -2476,6 +2479,10 @@ void RasterizerSceneGLES3::render_scene(const Ref<RenderSceneBuffers> &p_render_
 		if (render_data.environment.is_null() || (render_data.environment.is_valid() && !environment_get_fog_enabled(render_data.environment))) {
 			spec_constant_base_flags |= SceneShaderGLES3::DISABLE_FOG;
 		}
+
+		if (render_data.environment.is_null() || (render_data.environment.is_valid() && environment_get_fog_enabled(render_data.environment) && environment_get_fog_mode(render_data.environment) != RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH)) {
```

Comment:
The shader changes look correct, but this conditional needs to change to match. 

Right enabling depth fog requires:
1. There to be no environment or
2. there is an environment, fog is enabled, and fog is not set to depth mode

It should likely be simplified to only check if environment is valid and if fog mode is mode depth.